Taarafo.Portal.Web/Services/Foundations/Posts/PostService.Validations.cs

```diff
@@ -22,6 +40,27 @@ public static void ValidatePostId(Guid postId)
             Message = "Id is required"
         };
 
+        private static dynamic IsInvalid(string text) => new
+        {
+            Condition = String.IsNullOrWhiteSpace(text),
+            Message = "Text is required"
+        };
+
+        private static dynamic IsInvalid(DateTimeOffset date) => new
+        {
+            Condition = date == default,
+            Message = "Date is required"
+        };
+
+        private static dynamic IsNotSame(
+            DateTimeOffset firstDate,
+            DateTimeOffset secondDate,
+            string secondDateName) => new
+        {
+            Condition = firstDate != secondDate,
+            Message = $"Date is not the same as {secondDateName}"
+        };
```

Comment:
You don't need logical validations at the client side. Only structural.

Reply:
@hassanhabib thx. I removed this one. Please look again if you can find any improvements or missing tests that need to be handled.